docs/hugo/content/contributing/add-a-new-code-generated-resource/review-the-generated-resource.md

```diff
@@ -0,0 +1,127 @@
+
+```yaml
+  - group: documentdb
+    name: Location  # This type is subsequently flattened into NamespacesTopics_Spec
+    property: ProvisioningState
+    remove: true
+    because: This property should have been marked readonly but wasn't.
+```
+
+## Check: secrets generated by Azure
+
+Some resources, such as `microsoft.storage/accounts` and `microsoft.documentdb/databaseAccounts`, have keys and endpoints generated by
+Azure. Unfortunately, there is no good way to automatically detect these in the specs read by the code generator.
+
+**You must manually identify if the resource you are adding has keys (or endpoints) which users will want exported to a secret store**.
+
+If the resource in question _does_ have Azure generated secrets or endpoints, add those endpoints to the `azure-arm.yaml` configuration file
+by specifying `$azureGeneratedSecrets`.
+
+Our example resource above does not have any Azure generated secrets. As mentioned above, `microsoft.documentdb/databaseAccounts` has
+Azure generated secrets. Here is the snippet from the configuration file showing how they were configured:
+
+```yaml
+  documentdb:
+    2021-05-15:
+      DatabaseAccount:
+        $export: true
+        $azureGeneratedSecrets:
+          - PrimaryMasterKey
+          - SecondaryMasterKey
+          - PrimaryReadonlyMasterKey
+          - SecondaryReadonlyMasterKey
+          - DocumentEndpoint
+```
+
+Since these properties are manually configured, they must also be retrieved from Azure manually. This is done using
+the extension framework supported by the operator. Use the [documentdb secrets extension](https://github.com/Azure/azure-service-operator/blob/main/v2/api/documentdb/customizations/database_account_extensions.go)
+as a template for authoring an extension for your resource. Most extensions go into the `customizations` package found at the group level.
+
+<!-- TODO: Do we have a link to give more details about extensions and how they work? -->
+
+## Check: consumption of secrets
+
+Some resources take secrets as input from the user, for example when creating a virtual machine the user may supply a `pasword`.
+This is a `string` field but because it represents a secret value we don't want to just stick it on the `Spec` as a string.
+It needs to be a `genruntime.SecretReference` (a reference to a secret) instead, so that users can safely supply their
+secret to the operator without it being accessible via plain-text.
+
+In most cases, the Swagger/OpenAPI document will annotate these properties with an extension `x-ms-secret`, which indicates
+that value is a secret. The ASO code-generator uses that annotation to automatically transform the annotated field to
+a `genruntime.SecretReference`.
+
+In some cases, the resources Swagger specification is missing this annotation. If that happens, use an override in
+our `azure-arm.yaml` to force a particular property to be treated as a secret. For example, under `synapse`:
+
+```yaml
+WorkspaceProperties:
```

Comment:
The naming here, is this some convention where it's <resource>Properties? Or is this embedded within the workspace resource? Maybe clarify this.